File: Kth Missing Positive Number in a Sorted Array.java
class Solution {
    public int kthMissing(int[] arr, int k) {
        // code here
        int lo=0, hi = arr.length-1;
        int ans = arr.length + k;
        while(lo <= hi){
            int mid = (lo+hi)/2;
            if(arr[mid] > mid + k){
                hi = mid - 1;
                ans = mid + k;
            }
            else {
                lo = mid + 1;
            }
        }
        return ans;
    }
}

File: Kth Missing Positive Number in a Sorted Array.cpp
class Solution {
  public:
    int kthMissing(vector<int> &arr, int k) {
        // code here
         int n=arr.size();
        int ans=n;
        int start=0,end=n-1;
        while(start<=end){
            int mid=start+(end-start)/2;
            if(arr[mid]-mid-1>=k){
                ans=mid;
                end=mid-1;
            }
            else
            start=mid+1;
            
        }
        return ans+k;
    }
};
